### PR TITLE
Update dev dependencies

### PR DESCRIPTION
**Pull request by plmercereau**
Notably
- Vite 3 & plugins
- Vitest
- Turborepo
- Husky (probably solving [this issue](https://github.com/pnpm/pnpm/issues/4657))
- Faker 7

Remove unused dependencies

### CODE REVIEW

File: config/vite.lib.config.js [hunk 1]
```diff
@@ -18,13 +18,7 @@ export default defineConfig({
...
     dts({
       exclude: ['**/*.spec.ts', '**/*.test.ts', '**/tests/**'],
-      afterBuild: () => {
-        const types = fs.readdirSync(path.join(PWD, 'dist/src'))
-        types.forEach((file) => {
-          fs.renameSync(path.join(PWD, 'dist/src', file), path.join(PWD, 'dist', file))
-        })
-        fs.rmdirSync(path.join(PWD, 'dist/src'))
-      }
+      entryRoot: 'src'
```
Comment (plmercereau) [inline config/vite.lib.config.js:21]: new `entryRoot` param in the latest dts plugin version that replaces our `afterBuild` script

File: turbo.json [hunk 2]
```diff
@@ -3,24 +3,55 @@
...
+        "dist/**",
+        "umd/**",
+        "build/**",
+        ".next"
+      ]
     },
     "@nhost-examples/react-apollo#build": {
-      "dependsOn": ["^build", "$VITE_NHOST_SUBDOMAIN", "$VITE_NHOST_REGION"],
-      "outputs": ["dist/**"]
+      "dependsOn": [
```
Comment (plmercereau) [inline turbo.json:17]: the new version of turborepo introduces an `env` parameter

File: examples/react-apollo/cypress/e2e/3-sign-in/email-password.cy.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import totp from 'totp-generator'
 
-import faker from '@faker-js/faker'
+import { faker } from '@faker-js/faker'
```
Comment (plmercereau) [inline examples/react-apollo/cypress/e2e/3-sign-in/email-password.cy.ts:3]: the new version of `faker`  doesn't have a default export anymore

File: config/jest.config.base.js [hunk 4]
```diff
@@ -1,17 +0,0 @@
-module.exports = {
```
Comment (plmercereau) [inline config/jest.config.base.js:1]: we are not using jest anymore

File: config/vite.lib.config.js (hunk 1 above)
Comment (szilarddoro) [reply config/vite.lib.config.js:21]: Much cleaner. I wish I saw this prop when creating the config 🤦🏼

**Review by szilarddoro — APPROVED** [review]
Amazing. Thanks, @plmercereau!

Too bad that [Dependabot doesn't support PNPM](https://github.com/dependabot/dependabot-core/issues/1736#issuecomment-1272772849) yet...

**Comment by plmercereau** [comment]
> Amazing. Thanks, @plmercereau!
> 
> Too bad that [Dependabot doesn't support PNPM](https://github.com/dependabot/dependabot-core/issues/1736#issuecomment-1272772849) yet...

Yup, too bad :/
You didn't miss the `entryRoot` option, it wasn't existing at this time :)

**Pull request merged**